types/greenlock-express/index.d.ts

```diff
@@ -0,0 +1,28 @@
+// Type definitions for greenlock-express 4.0
+// Project: https://git.rootprojects.org/root/greenlock-express.js.git
+// Definitions by: Joseph Hendrix <https://github.com/hendrixjoseph>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+import { RequestListener } from 'http';
+
+export interface opts {
+    packageRoot: string;
+    maintainerEmail: string;
+    configDir: string;
+    cluster: boolean;
+    workers?: number;
+}
+
+export interface serve {
+    ready: (app?: RequestListener) => serve;
+    master: (app?: RequestListener) => serve;
+    serve: (app: RequestListener) => void;
+}
+
+export interface greenlock {
+    init: (o: (opts | (() => opts))) => serve;
+}
+
+declare const gl: greenlock;
+
+export default gl;
```

Comment:
```suggestion
export function init(o: (opts | (() => opts))): serve;
```
or something along those lines

Reply:
It seems when I replace the default export with just the function:
```typescript
export default function init(o: (opts | (() => opts))): serve;
```
it expects that the import itself is the function, rather than having an `init` function. e.g.:

e.g. it expects:
```typescript
import greenlock from 'greenlock-express';

greenlock(o);
```
rather than:
```typescript
import greenlock from 'greenlock-express';

greenlock.init(o);
```
---
I could instead declare the const like follows:
```typescript
declare const greenlock: {
    init: (o: (opts | (() => opts))) => serve;
};

export = greenlock;
```
if I wanted to breakout the function definition on it's own, I could also:
```typescript
declare function init(o: (opts | (() => opts))): serve;

declare const greenlock: {
    init: typeof init
};

export = greenlock;
```

Reply:
sorry for possible confusion, here is one approach. The module just exports `init` function and DT types could add types symbols to be exported as well (options for example).
so the declaration file:
```ts
import type { RequestListener } from 'node:http';

export interface Options {
    packageRoot: string;
    maintainerEmail: string;
    configDir: string;
    cluster: boolean;
    workers?: number;
}

export interface Serve {
    ready: (app?: RequestListener) => Serve;
    master: (app?: RequestListener) => Serve;
    serve: (app: RequestListener) => void;
}

export function init(o: (Options | (() => Options))): Serve;
```
while the tests:
```ts
import express = require("express");
import type { RequestListener } from 'node:http';
import greenlock = require('greenlock-express');

const o: greenlock.Options = {
    packageRoot: './',
    maintainerEmail: 'user@example.com',
    configDir: './greenlock.d',
    cluster: false
};

const expressApp = express();

// $ExpectType Serve
greenlock.init(o);
...
```
(notice changed notation, CamelCase for symbols as per JS conventions)

Reply:
no worries.

I've updated the index type file and the tests; I like this better than my initial approach, too.

I also tested the types in a separate project using the `import greenlock from 'greenlock-express';` and it works with that, too.

Quick question - is there a particular reason you used `import type { RequestListener } from 'node:http';` instead of `import type { RequestListener } from 'http';` without the `node:` part?

Reply:
using scoped packages is a convention, not requirement. It solves a problem like with NPM package 'buffer': https://www.npmjs.com/package/buffer, removing disambiguate from imports and types resolution. thx!